drivers/serial/uart_ns16550.c

```diff
@@ -841,7 +841,11 @@ static int uart_ns16550_init(const struct device *dev)
 		data->async.tx_dma_params.dma_cfg.head_block =
 			&data->async.tx_dma_params.active_dma_block;
 #if defined(CONFIG_UART_NS16550_INTEL_LPSS_DMA)
+#if UART_NS16550_IOPORT_ENABLED
 		if (!dev_cfg->io_map) {
+#else
+		{
+#endif
```

Comment:
You can remove the `{` on the if line, and leave the other one outside of the ifdef block.